poh/src/poh_recorder.rs

```diff
@@ -47,9 +47,6 @@ pub const MAX_GRACE_SLOTS: u64 = 2;
 
 #[derive(Error, Debug, Clone)]
 pub enum PohRecorderError {
-    #[error("invalid calling object")]
-    InvalidCallingObject,
```

Comment:
git historian here :) 

seems this variant is added at #1604 and its use is removed at #1747 without removing this variant. i think this is simply overlook.

```patch
$ git log -p -S InvalidCallingObject | grep -C5 -E '((^[^+-\ ])|InvalidCallingObject)'
commit d831c5dcc957817a61de371e29106719bf88294c
Author: Rob Walker <rob@solana.com>
Date:   Thu Nov 8 12:55:23 2018 -0800

    remove dead poh code (#1747)

diff --git a/src/poh_recorder.rs b/src/poh_recorder.rs
index bcbd4ea030f..bfee2039d0d 100644
--- a/src/poh_recorder.rs
+++ b/src/poh_recorder.rs
@@ -18,15 +18,10 @@ pub enum PohRecorderError {

 #[derive(Clone)]
--
     }

     pub fn record(&self, mixin: Hash, txs: Vec<Transaction>) -> Result<()> {
-        if self.is_virtual {
-            return Err(Error::PohRecorderError(
-                PohRecorderError::InvalidCallingObject,
-            ));
-        }
         // Register and send the entry out while holding the lock.
         // This guarantees PoH order and Entry production and banks LastId queue is the same.
         let mut poh = self.poh.lock().unwrap();
--
+        let mut poh_recorder = PohRecorder::new(bank, entry_sender, last_id, None);

         //send some data
         let h1 = hash(b"hello world!");

commit 55833e20b14f09d44539af5afa18fb09af85ba28
Author: carllin <wumu727@gmail.com>
Date:   Thu Oct 25 14:56:21 2018 -0700

    Create Poh Service (#1604)

    * Create new Poh Service, replace tick generation in BankingStage

diff --git a/src/poh_recorder.rs b/src/poh_recorder.rs
index c525759bda3..96acf66e1f0 100644
--- a/src/poh_recorder.rs
+++ b/src/poh_recorder.rs
@@ -12,11 +12,14 @@ use transaction::Transaction;

 #[derive(Debug, PartialEq, Eq, Clone)]
 pub enum PohRecorderError {
+    InvalidCallingObject,
     MaxHeightReached,
 }

 #[derive(Clone)]
 pub struct PohRecorder {
--
     }

     pub fn record(&self, mixin: Hash, txs: Vec<Transaction>) -> Result<()> {
+        if self.is_virtual {
+            return Err(Error::PohRecorderError(
+                PohRecorderError::InvalidCallingObject,
+            ));
+        }
         // Register and send the entry out while holding the lock.
         // This guarantees PoH order and Entry production and banks LastId queue is the same.
         let mut poh = self.poh.lock().unwrap();

```